source/exe/main_common.cc

```diff
@@ -144,7 +148,12 @@ std::string MainCommon::hotRestartVersion(uint64_t max_num_stats, uint64_t max_s
 // and MainCommon can be merged. The current theory is that only Google calls this.
 int main_common(OptionsImpl& options) {
   try {
-    MainCommonBase main_common(options);
+    Event::RealTimeSystem real_time_system_;
```

Comment:
As long as you are in here, is Google still using this per the TODO above? Can we just delete it?

Reply:
Sadly, yes.  I don't know how widely it's used, but there's a binary that goes through this pathway that looks like our most general Google envoy binary (which doesn't mean it's widely used; the use cases I'm aware of create their own binary).  @jmarantz is I think the expert in this space, but I think he's also out on medical leave.  So I don't think we can nuke this as part of this CL.

Reply:
OK. :(

Reply:
Assuming nobody else is using this: I can clean this up; I created an internal ticket to fix the caller, and then I'll follow up with a PR that removes it here.

Reply:
sorry i have an  unpushed branch to do this, but I did not finish and don't have access to it now. Feel free to take over or I could progress on that later this week.

Reply:
I've eliminated the internal caller as of now. Since you already have a WIP for the upstream change, I'll leave that for you. Thanks!